bingo.js

```diff
@@ -602,9 +675,119 @@ function setSquareColor(square, colorClass)
 {
 	ALL_COLOURS.forEach(c => square.removeClass(c));
 	square.addClass(colorClass);
+	updateCurrentSheetProgress();
```

Comment:
Saving of progress in `setSquareColor` means that when "Never" goals are highlighted on sheet generation, the "progress" is saved before user clicked any squares. This means that a player can lose their real progress by just clicking on "Generate New Seed", which is undesirable.

Programmatic and human setting of square colours needs to be differentiated.